pandas/tests/indexes/object/test_indexing.py

```diff
@@ -114,6 +114,13 @@ def test_get_indexer_non_unique_np_nats(self, np_nat_fixture, np_nat_fixture2):
             tm.assert_numpy_array_equal(missing, expected_missing)
         # dt64nat vs td64nat
         else:
+            try:
+                np_nat_fixture == np_nat_fixture2
+            except (TypeError, OverflowError):
+                # Numpy will raise on uncomparable types, like
+                # np.datetime64('NaT', 'Y') and np.datetime64('NaT', 'ps')
+                # https://github.com/numpy/numpy/issues/22762
+                return
```

Comment:
Is this failure only on numpy dev, i.e. numpy > 1.23.x? Could this be better by having a `p_version_lte1p23 = _nlv <= Version("1.23")` in copat.py and then do:

```python
if p_version_lte1p23:
    return

np_nat_fixture == np_nat_fixture2
...
```
?

That would make this easier to find and remove when we drop support for numpy 1.23 at some point in the future).

Reply:
true, though comparisions between `timedelta64 NaT` and `datetime64 NaT` will still work

maybe that can be split out - I'll do that, thanks!

Reply:
+1

Reply:
aybe in a followup, so we can get the CI to work again?